pkg/network/protocols/postgres/telemetry.go

```diff
@@ -97,3 +98,131 @@ func (t *Telemetry) Count(tx *ebpf.EbpfEvent, eventWrapper *EventWrapper) {
+	pgKernelMsgCountPlainPrev pgebpf.PostgresKernelMsgCount
+	// pgKernelMsgCountPrev			save the last counters observed from the kernel, TLS messages
+	pgKernelMsgCountTlsPrev pgebpf.PostgresKernelMsgCount
+}
+
+// newPgKernelTelemetry this is the Postgres message counter store.
+func newPgKernelTelemetry() *PgKernelTelemetry {
+	metricGroup := libtelemetry.NewMetricGroup("usm.postgres_kernel", libtelemetry.OptStatsd)
+	pgKernelTel := &PgKernelTelemetry{
+		metricGroup:             metricGroup,
+		pgMessagesCountNotFound: libtelemetry.NewTLSAwareCounter(metricGroup, "messages_count_not_found"),
+		pgMessagesCountAboveMax: libtelemetry.NewTLSAwareCounter(metricGroup, "messages_count_above_max"),
+		pgMessagesCountErrors:   libtelemetry.NewTLSAwareCounter(metricGroup, "messages_count_errors"),
+	}
+	pgKernelTel.createMessagesCountBuckets(metricGroup)
+
+	return pgKernelTel
+}
+
+func (t *PgKernelTelemetry) createMessagesCountBuckets(metricGroup *libtelemetry.MetricGroup) {
+	for bucketIndex := range t.pgMessagesCountBuckets {
+		t.pgMessagesCountBuckets[bucketIndex] = libtelemetry.NewTLSAwareCounter(metricGroup,
+			"messages_count_bucket_"+(strconv.Itoa(bucketIndex+1)))
+	}
+}
+
+// update the Postgres message counter store with new counters from the kernel.
+// return if nothing to add, in this case also metric group summary ignores unmodified metrics.
+func (t *PgKernelTelemetry) update(kernMsgCounts *pgebpf.PostgresKernelMsgCount, isTLS bool) {
+	if kernMsgCounts == nil {
+		return
+	}
+	if isEmptyCounts(kernMsgCounts) {
+		log.Debugf("postgres kernel telemetry empty.")
+		return
+	}
+	delta := t.getDiff(kernMsgCounts, isTLS)
+	if isEmptyCounts(delta) {
+		log.Debugf("postgres kernel telemetry, no difference with previous counters.")
+		return
+	}
+	t.pgMessagesCountNotFound.Add(int64(delta.Messages_not_found), isTLS)
+	t.pgMessagesCountAboveMax.Add(int64(delta.Messages_above_max), isTLS)
+	t.pgMessagesCountErrors.Add(int64(delta.Messages_count_err), isTLS)
+	for bucketIndex := range t.pgMessagesCountBuckets {
+		t.pgMessagesCountBuckets[bucketIndex].Add(int64(delta.Messages_count_buckets[bucketIndex]), isTLS)
+	}
+	if isTLS {
+		t.pgKernelMsgCountTlsPrev = *kernMsgCounts
+	} else {
+		t.pgKernelMsgCountPlainPrev = *kernMsgCounts
+	}
+	s := t.metricGroup.Summary()
+	log.Debugf("postgres kernel telemetry, isTLS=%t, summary: %s", isTLS, s)
+}
+
+func (t *PgKernelTelemetry) getDiff(kernMsgCounts *pgebpf.PostgresKernelMsgCount, isTLS bool) *pgebpf.PostgresKernelMsgCount {
+	if isTLS {
+		delta := subtractCounts(kernMsgCounts, t.pgKernelMsgCountTlsPrev)
+		return delta
+	} else {
+		delta := subtractCounts(kernMsgCounts, t.pgKernelMsgCountPlainPrev)
+		return delta
+	}
+}
+
+// subtractCounts create a new PostgresKernelMsgCount by subtracting the counts of 'b' from 'a'.
+// calculate only increasing counters, otherwise subtraction gives unreasonably large values for unsigned integers.
```

Comment:
> calculate only increasing counters, otherwise subtraction gives unreasonably large values for unsigned integers.

That's an error, you shouldn't be silencing this error

Reply:
done